lib/Runtime/Debug/TTSnapObjects.h

```diff
@@ -1075,6 +1080,70 @@ namespace TTD
 #if ENABLE_SNAPSHOT_COMPARE
         void AssertSnapEquiv_SnapProxyInfo(const SnapObject* sobj1, const SnapObject* sobj2, TTDCompareMap& compareMap);
 #endif
+    
+
+        //////////////////
+
+        struct SnapGeneratorFunctionInfo {
+            TTD_PTR_ID scriptFunction; // pointer to GeneratorVirtualScriptFunction
+            bool isAnonymousFunction;
+        };
+
+        struct SnapGeneratorVirtualScriptFunctionInfo : SnapScriptFunctionInfo {
+            TTD_PTR_ID realFunction; //pointer to JavascriptGeneratorFunction
+
+        };
+        
+        struct SnapAsyncFunctionInfo {
+
+        };
+       
+        struct SnapGeneratorInfo {
+            int generatorPrototype; // 0 == unknown, 1 == nullType, 2 == generatorPrototype
+            Js::RegSlot frame_slotCount;
+            TTDVar* frame_slotArray;
+            uint32 state; // enum value of JavascriptGenerator.GeneratorState
+            TTD_PTR_ID scriptFunction;
+            byte arguments_callInfo[sizeof(Js::CallInfo)];
```

Comment:
Just checking: while we do have a goal of windows/xplat trace compatibility, we don't have a goal of cross-architecture compatibility, right? So potential endianness issues aren't important?

Reply:
That is correct, cross-architecture is explicitly not supported.